benchmark/vendor.h

```diff
@@ -0,0 +1,112 @@
+#ifndef BENCHMARK_VENDOR_H_
+#define BENCHMARK_VENDOR_H_
+
+#include <stdint.h>
+
+#include <string>
+#include <string_view>
+
+#include "absl/strings/substitute.h"
+
+#include "tachyon/base/flag/flag_value_traits.h"
+#include "tachyon/base/logging.h"
+
+namespace tachyon {
+namespace benchmark {
+
+class Vendor {
+ public:
+  enum Value : uint32_t {
+    // clang-format off
+    kTachyonCPU  = 1 << 1,
+    kTachyonGPU  = 1 << 2,
+    kArkworks    = 1 << 3,
+    kBellman     = 1 << 4,
+    kScrollHalo2 = 1 << 5,
+    kPseHalo2    = 1 << 6,
+    kHorizen     = 1 << 7,
+    kPlonky3     = 1 << 8,
+    // clang-format on
+  };
+
+  Vendor() = default;
+  explicit constexpr Vendor(Value v) : value_(v) {}
```

Comment:
In this way, `benchmark::Vendor(benchmark::kTachyonCPU)` can be simplified with `benchmark::Vendor::TachyonCPU()`.

```suggestion
  constexpr static TachyonCPU() { return Vendor(kTachyonCPU); }
  // ...
  
 private:
  explicit constexpr Vendor(Value v) : value_(v) {}
```